### PR TITLE
Run CI on Windows

### LINKED ISSUE #1262: Get the CI working on Windows
In the interest of sanity, this should only apply to the JVM builds. It's going to require some fixes to `IOAppSpec` as well. At the very least:

```diff
diff --git a/core/jvm/src/test/scala/cats/effect/IOAppSpec.scala b/core/jvm/src/test/scala/cats/effect/IOAppSpec.scala
index f49c0bec..04d8fa29 100644
--- a/core/jvm/src/test/scala/cats/effect/IOAppSpec.scala
+++ b/core/jvm/src/test/scala/cats/effect/IOAppSpec.scala
@@ -35,14 +35,14 @@ class IOAppSpec extends Specification {
     "evaluate and print hello world" in {
       val h = java(HelloWorld, Nil)
       h.awaitStatus() mustEqual 0
-      h.stdout() mustEqual "Hello, World!\n"
+      h.stdout() mustEqual s"Hello, World!${System.lineSeparator}"
     }

     "pass all arguments to child" in {
       val expected = List("the", "quick", "brown", "fox jumped", "over")
       val h = java(Arguments, expected)
       h.awaitStatus() mustEqual 0
-      h.stdout() mustEqual expected.mkString("", "\n", "\n")
+      h.stdout() mustEqual expected.mkString("", System.lineSeparator, System.lineSeparator)
     }

     "run finalizers on TERM" in {
```

The advantage to testing on Windows, aside from just making it possible for Windows users to contribute without working around failing tests, is the upcoming character encoding unit tests for `Console`.

### PR DESCRIPTION
**Pull request by vasilmkd**
Resolves #1262.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -69,21 +70,25 @@ ThisBuild / githubWorkflowBuild := Seq(
   WorkflowStep.Run(
     List("example/test-jvm.sh ${{ matrix.scala }}"),
     name = Some("Test Example JVM App Within Sbt"),
-    cond = Some("matrix.ci == 'ciJVM'")),
+    cond = Some("matrix.ci == 'ciJVM' && matrix.os == 'ubuntu-latest'")),
```
Comment (djspiewak) [inline build.sbt]:
```suggestion
    cond = Some("matrix.ci == 'ciJVM' && matrix.os == PrimaryOS")),
```

File: build.sbt [hunk 2]
```diff
@@ -69,21 +70,25 @@ ThisBuild / githubWorkflowBuild := Seq(
...
 
   WorkflowStep.Run(
     List("example/test-js.sh ${{ matrix.scala }}"),
     name = Some("Test Example JavaScript App Using Node"),
-    cond = Some("matrix.ci == 'ciJS'")))
+    cond = Some("matrix.ci == 'ciJS' && matrix.os == 'ubuntu-latest'")))
```
Comment (djspiewak) [inline build.sbt]:
```suggestion
    cond = Some("matrix.ci == 'ciJS' && matrix.os == PrimaryOS")))
```

File: core/jvm/src/test/scala/cats/effect/IOAppSpec.scala [hunk 3]
```diff
@@ -35,50 +35,56 @@ class IOAppSpec extends Specification {
...
-
-      // we have to resort to this convoluted approach because Process#destroy kills listeners before killing the process
-      val test = File.createTempFile("cats-effect", "finalizer-test")
-      def readTest(): String = {
-        val reader = new BufferedReader(new FileReader(test))
-        try {
-          reader.readLine()
-        } finally {
-          reader.close()
+      if (System.getProperty("os.name").contains("Windows")) {
```
Comment (djspiewak) [inline core/jvm/src/test/scala/cats/effect/IOAppSpec.scala]: Older versions of windows don't use capitalization. This is rather annoying. The workaround:

```suggestion
      if (System.getProperty("os.name").toLowerCase.contains("windows")) {
```

File: core/jvm/src/test/scala/cats/effect/IOAppSpec.scala [hunk 4]
```diff
@@ -35,50 +35,56 @@ class IOAppSpec extends Specification {
...
+        // The jvm cannot gracefully terminate processes on Windows, so this
+        // test cannot be carried out properly. Same for testing IOApp in sbt.
+        ok
```
Comment (djspiewak) [inline core/jvm/src/test/scala/cats/effect/IOAppSpec.scala]:
```suggestion
        skipped
```

This is a bit more accurate

**Review by djspiewak — CHANGES_REQUESTED** [review]

File: core/jvm/src/test/scala/cats/effect/IOAppSpec.scala (hunk 4 above)
Comment (vasilmkd) [reply core/jvm/src/test/scala/cats/effect/IOAppSpec.scala]: Skipped doesn't compile as it is not a `MatchResult`...

File: build.sbt (hunk 1 above)
Comment (vasilmkd) [reply build.sbt]: Done.

File: core/jvm/src/test/scala/cats/effect/IOAppSpec.scala (hunk 3 above)
Comment (vasilmkd) [reply core/jvm/src/test/scala/cats/effect/IOAppSpec.scala]: Done.

File: core/jvm/src/test/scala/cats/effect/IOAppSpec.scala (hunk 4 above)
Comment (djspiewak) [reply core/jvm/src/test/scala/cats/effect/IOAppSpec.scala]: Welp. Apparently it's `Skipped()`

Comment (vasilmkd) [reply core/jvm/src/test/scala/cats/effect/IOAppSpec.scala]: Also not, that's the same type 😆

Comment (djspiewak) [reply core/jvm/src/test/scala/cats/effect/IOAppSpec.scala]: Okay apparently I don't know specs very well. I'll fiddle with it later :-P

Comment (vasilmkd) [reply core/jvm/src/test/scala/cats/effect/IOAppSpec.scala]: Does anyone? 😄

**Pull request merged**